src/CORS/src/Microsoft.AspNetCore.Cors/Infrastructure/CorsMiddleware.cs

```diff
@@ -149,6 +150,12 @@ private async Task InvokeCore(HttpContext context, ICorsPolicyProvider corsPolic
             }
         }
 
+        internal string ResolveCorsPolicyName(HttpContext context)
+        {
+            var endpoint = context.GetEndpoint();
+            return endpoint?.Metadata.GetMetadata<IEnableCorsAttribute>()?.PolicyName ?? _corsPolicyName;
+        }
```

Comment:
re: `IDisableCorsAttribute` - what I'd imagine is that we need to add a marker interface that both of these implement, and check the type of the most significant one.

I would expect `IDisableCorsAttribute` to work, it's a feature MVC supports